### PR TITLE
ZER-58 Add validation and error messaging to the registration page

### PR DESCRIPTION
**Pull request by aboriiisova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/user/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class User::RegistrationsController < Devise::RegistrationsController
+  # Override the action you want here.
+  protect_from_forgery with: :exception
+  before_action :sign_up_params, only [:create]
+
+  def create
+    super 
+  end
```
Comment (maksymst) [inline app/controllers/user/registrations_controller.rb:8]: What the purpose of this change?

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -24,7 +24,7 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+  config.mailer_sender = 'annaborysova99@gmail.com'
```
Comment (maksymst) [inline config/initializers/devise.rb:27]: Put here an environment variable

File: config/locales/en.yml [hunk 3]
```diff
@@ -154,3 +154,10 @@ en:
       user:
         email: "Email"
         password: "Password"
+        password_confirmation: "Re-password"
+        first_name: "First name"
+        last_name: "Last name"
+          #blank: "Please input a last name"
+          #invalid: "Please input a correct last name"
```
Comment (maksymst) [inline config/locales/en.yml:161]: do not commit commented out code

File: config/locales/uk.yml [hunk 4]
```diff
@@ -54,4 +54,18 @@ uk:
...
+          #invalid: "Може містити цифри. символи. великі та малі букви та бути не менше 8 символів"
+        password_confirmation: "Повторний пароль"
+          #blank: "Вкажіть пароль ще раз"
+          #confirmation: "Паролі не співпадають"
+        first_name: "Ім'я"
+          #blank: "Вкажіть ім'я"
+          #invalid: "Може містити тільки букви"
+        last_name: "Прізвище"
+          #blank: "Вкажіть прізвище"
+          #invalid: "Може містити тільки букви"
```
Comment (maksymst) [inline config/locales/uk.yml:70]: same as above

File: db/migrate/20210518085148_add_confirmable_to_devise.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class AddConfirmableToDevise < ActiveRecord::Migration[6.1]
+  def up
+    add_column :users, :confirmation_token, :string
+    add_column :users, :confirmed_at, :datetime
+    add_column :users, :confirmation_sent_at, :datetime
```
Comment (maksymst) [inline db/migrate/20210518085148_add_confirmable_to_devise.rb:5]: replace with change_table

File: db/migrate/20210518085148_add_confirmable_to_devise.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    # add_column :users, :unconfirmed_email, :string # Only if using reconfirmable
```
Comment (maksymst) [inline db/migrate/20210518085148_add_confirmable_to_devise.rb:6]: do not commit commented out code

File: db/migrate/20210426125453_devise_create_users.rb [hunk 7]
```diff
@@ -21,11 +21,11 @@ def change
...
-      ## Confirmable
-      # t.string   :confirmation_token
-      # t.datetime :confirmed_at
-      # t.datetime :confirmation_sent_at
-      # t.string   :unconfirmed_email # Only if using reconfirmable
+      # Confirmable
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      #t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (maksymst) [inline db/migrate/20210426125453_devise_create_users.rb:28]: do not commit commented out code

**Review by maksymst — CHANGES_REQUESTED** [review]

File: config/initializers/devise.rb (hunk 2 above)
Comment (aboriiisova) [reply config/initializers/devise.rb]: Done

File: db/migrate/20210518085148_add_confirmable_to_devise.rb (hunk 5 above)
Comment (aboriiisova) [reply db/migrate/20210518085148_add_confirmable_to_devise.rb]: Fixed

**Review by haliapats — APPROVED** [review]

File: db/migrate/20210518085148_add_confirmable_to_devise.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+class AddConfirmableToDevise < ActiveRecord::Migration[6.1]
+  def change
+    change_table(:users) do |t|
+      t.column :confirmation_token, :string
+      t.column :confirmed_at, :datetime
+      t.column ::confirmation_sent_at, :datetime
```
Comment (maksymst) [inline db/migrate/20210518085148_add_confirmable_to_devise.rb:6]: fix this

File: config/locales/uk.yml [hunk 9]
```diff
@@ -55,3 +55,8 @@ uk:
       user:
         email: "Логін"
         password: "Пароль"
+        password_confirmation: "Повторний пароль"
+        first_name: "Ім'я"
+        last_name: "Прізвище"
+         
+
```
Comment (maksymst) [inline config/locales/uk.yml:62]: remove empty lines

File: config/locales/en.yml [hunk 10]
```diff
@@ -154,3 +154,9 @@ en:
       user:
         email: "Email"
         password: "Password"
+        password_confirmation: "Re-password"
+        first_name: "First name"
+        last_name: "Last name"
+
+
+
```
Comment (maksymst) [inline config/locales/en.yml:162]: remove empty lines

File: config/initializers/devise.rb [hunk 11]
```diff
@@ -24,7 +24,8 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+  sender_mail = 'annaborysova99@gmail.com'
+  config.mailer_sender = sender_mail
```
Comment (maksymst) [inline config/initializers/devise.rb:28]: still not environment variable

File: db/migrate/20210518085148_add_confirmable_to_devise.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+class AddConfirmableToDevise < ActiveRecord::Migration[6.1]
+  def change
+    change_table(:users) do |t|
```
Comment (maksymst) [inline db/migrate/20210518085148_add_confirmable_to_devise.rb:3]: after changing this migration you've lost some previous changes

**Review by maksymst — CHANGES_REQUESTED** [review]

**Pull request closed without merging**